uptime/checker.go

```diff
@@ -0,0 +1,416 @@
+			if err != nil {
+				log.Errorw("cannot list member multi addrs", "actor", toCheckActorID, "err", err)
+				continue
+			}
+
+			log.Debugw("member info", "actor", toCheckActorID, "addrs", addrs)
+
+			u.CheckMember(toCheckActorID, addrs)
+		}
+
+		time.Sleep(time.Duration(5) * time.Second)
+	}
+
+	return nil
+}
+
+func (u *UptimeChecker) monitorCheckerNodes(ctx context.Context) error {
+	for {
+		if u.IsStop() {
+			break
+		}
+
+		state, err := Load(ctx, u.api, u.actorAddress, u.self)
+		if err != nil {
+			log.Errorw("cannot load state", "err", err)
+			continue
+		}
+
+		listToCheck, err := state.ListCheckers()
+		if err != nil {
+			log.Errorw("cannot list members", "err", err)
+			continue
+		}
+
+		for _, toCheckPeerID := range listToCheck {
+			addrs, err := state.ListCheckerMultiAddrs(toCheckPeerID)
+			if err != nil {
+				log.Errorw("cannot list member multi addrs", "peer", toCheckPeerID, "err", err)
+				continue
+			}
+
+			log.Debugw("toCheckPeerID addrs", "toCheckPeerID", toCheckPeerID, "addrs", addrs)
+
+			err = u.CheckChecker(ctx, toCheckPeerID, addrs)
+			if err != nil {
+				log.Errorw("cannot check checker", "peer", toCheckPeerID, "err", err)
+			}
+		}
+
+		time.Sleep(time.Duration(5) * time.Second)
+	}
+
+	return nil
+}
+
+func (u *UptimeChecker) execSync(ctx context.Context, method uint32, from address.Address, params []byte) error {
+	smsg, err := u.execSyncNoWait(ctx, method, from, params)
+	if err != nil {
+		return err
+	}
+
+	log.Infow("waiting for message to execute...")
+	return u.wait(ctx, smsg)
+}
+
+func (u *UptimeChecker) execSyncNoWait(ctx context.Context, method uint32, from address.Address, params []byte) (*chainTypes.SignedMessage, error) {
+	msg := &chainTypes.Message{
+		To:     u.actorAddress,
+		From:   from,
+		Value:  big.Zero(),
+		Method: abi.MethodNum(method),
+		Params: params,
+	}
+	return u.api.MpoolPushMessage(ctx, msg, nil)
+}
+
+func (u *UptimeChecker) wait(ctx context.Context, smsg *chainTypes.SignedMessage) (error) {
+	wait, err := u.api.StateWaitMsg(ctx, smsg.Cid(), 0)
+	if err != nil {
+		return err
+	}
+
+	// check it executed successfully
+	if wait.Receipt.ExitCode != 0 {
+		return fmt.Errorf("actor execution failed")
+	}
+
+	return nil
+}
+
+// Checks is up and also record the latency
+func (u *UptimeChecker) isUp(addrStr MultiAddr) UpInfo {
+	upInfo := UpInfo{
+		isOnline: false,
+		latency: uint64(0),
+		checkedTime: uint64(time.Now().Unix()),
+	}
+
+	addr, err := libp2pMultiaddr.NewMultiaddr(addrStr)
+	if err != nil {
+		log.Errorw("cannot parse multi addr", "addr", addr)
+		return upInfo
+	}
+
+	peer, err := peerstore.AddrInfoFromP2pAddr(addr)
+	if err != nil {
+		log.Errorw("cannot add multi addr", "addr", addr)
+		return upInfo
+	}
+
+	if err := u.node.Connect(context.Background(), *peer); err != nil {
+		log.Errorw("cannot connect to multi addr", "peer", peer.ID)
+		return upInfo
+	}
+
+	ch := u.ping.Ping(context.Background(), peer.ID)
```

Comment:
I think you are not computing the latency here. A simple thing that you can do (it may not be the most accurate but I think it works for now) is checking how much time the pint takes. You can do something like:
```go
now := time.Now()
// Wait for ping
// and if successful
latency := time.Since(now)
```